src/main/java/com/stuypulse/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,120 @@
+
+    // PIDController
+    private PIDController shooterController;
+
+    public Shooter() {
+        leftShooterMotor = new CANSparkMax(Constants.LEFT_SHOOTER_MOTOR_PORT, MotorType.kBrushless);
+        rightShooterMotor = new CANSparkMax(Constants.RIGHT_SHOOTER_MOTOR_PORT, MotorType.kBrushless);
+        middleShooterMotor = new CANSparkMax(Constants.MIDDLE_SHOOTER_MOTOR_PORT, MotorType.kBrushless);
+
+        leftShooterMotor.setInverted(true); //engineering said that the left motor is inverted
+
+        leftShooterEncoder = new NEOEncoder(leftShooterMotor.getEncoder());
+        rightShooterEncoder = new NEOEncoder(rightShooterMotor.getEncoder());
+        middleShooterEncoder = new NEOEncoder(middleShooterMotor.getEncoder());
+
+        feederMotor = new CANSparkMax(Constants.FEEDER_MOTOR_PORT, MotorType.kBrushless);
+
+        hoodSolenoid = new Solenoid(Constants.HOOD_SOLENOID_PORT);
+
+        shooterMotors = new SpeedControllerGroup(leftShooterMotor, rightShooterMotor, middleShooterMotor);
+
+        shooterController = new PIDController(Constants.SHOOTER_SHOOT_KP, Constants.SHOOTER_SHOOT_KI, Constants.SHOOTER_SHOOT_KD);
+    }
+
+    public double getError(double targetVelocity) {
+        return targetVelocity - getCurrentShooterVelocityInRPM();
+        // the erorr is calculated by subtracting the target velocity by the current velocity.
+    }
+    
+    public double getRawMaxShooterVelocity() {
+        return Math.max(leftShooterEncoder.getVelocity(), Math.max(middleShooterEncoder.getVelocity(), rightShooterEncoder.getVelocity())); 
+        // gets the max raw velocity of the shooter motors
+    }
+
+    public double getCurrentShooterVelocityInRPM() {
+        return getRawMaxShooterVelocity() * Constants.SHOOTER_VELOCITY_EMPIRICAL_MULTIPLER;
+        // returns the velocity of the shooter motors IN RPM by multiplying by the empirical factor. (TODO)
+    }
+
+    public double getScaledVelocity(double targetVelocity) {
+        return (shooterController.calculate(getError(targetVelocity), 0)) / Constants.SHOOTER_MAX_RPM;
+        // converts the output of the PID loop to the NEOEncoder unit (-1.0 to 1.0) by
+        // dividing the PIDOutput by the maximum RPM possible of the CANSparkMax motor.
+        // Sam said that this is already taken in account by the P value.
+
+        // the calculate method takes in a (double measuremnt, double setpoint) where 
+        // the measurement is the error and the setpoint is the target RPM.
+        // we want the error to be 0, which means the targetVelocity has been reached.
+    }
+
+    public void runShooter(double targetVelocity) {
+        shooterMotors.set(getScaledVelocity(targetVelocity));
+        // PID input = targetVeloctiy
+        // sets the encoder to the scaled velocity output of the PID loop.
+    }
+
+    public void stopShooter() {
+        shooterMotors.set(0.0);
+    }
+
+    public void runFeeder() {
+        feederMotor.set(1.0);
```

Comment:
should be a constant, and should also be PID tuned so that we have a constant velocuty feeding in

Reply:
by constant, i mean in `Constants.java`